state/state_test.go

```diff
@@ -4660,6 +4656,53 @@ func (s *StateSuite) TestWatchAPIHostPorts(c *gc.C) {
 	wc.AssertClosed()
 }
 
+func (s *StateSuite) TestWatchAPIHostPortsForAgents(c *gc.C) {
+	s.SetJujuManagementSpace(c, "mgmt01")
+
+	w := s.State.WatchAPIHostPortsForAgents()
+	defer statetesting.AssertStop(c, w)
+
+	// Initial event.
+	wc := statetesting.NewNotifyWatcherC(c, s.State, w)
+	wc.AssertOneChange()
+
+	mgmtHP := network.HostPort{
+		Address: network.Address{
+			Value:     "0.4.8.16",
+			Type:      network.IPv4Address,
+			Scope:     network.ScopeCloudLocal,
+			SpaceName: "mgmt01",
+		},
+		Port: 2,
+	}
+
+	err := s.State.SetAPIHostPorts([][]network.HostPort{{
+		mgmtHP,
+	}})
+	c.Assert(err, jc.ErrorIsNil)
```

Comment:
would this be better set up as doing this change, then AssertOneChange and then doing the next Set and AssertNoChange?
because of how notifications can get collapsed, etc, I have the feeling this isn't quite testing what you think it is testing.

Reply:
I understand. Will fix.